app/javascript/components/Pages/EmotionIntensity.js

```diff
@@ -29,23 +70,58 @@ const EmotionIntensity = ({data, setData, saveDataToDb, steps, service}) => {
     <h3 className='mt-3'>Select how intense the feeling was</h3>
   </Fragment>
 
-  const IntenseLine = () => <div>
-    <span className="under-convert">1</span>
-    <span className="under-convert">2</span>
-    <span className="under-convert">3</span>
-    <span className="under-convert">4</span>
-    <span className="under-convert">5</span>
-  </div>
+  const generateStyles = (value, selected, category) => {
+    let backgroundColor, borderColor;
+    switch(category) {
+      case "negative":
+        backgroundColor = 
+          value === 5 ? '#F18C59' : value === 4 ? '#F09D74' :
+          value === 3 ? '#F5C1A6' : value === 2 ? '#F7CDB8' :
+          value === 1 ? '#FADFD1' : 'transparent';
```

Comment:
Can you please explain what is going on here?

Reply:
@rogergraves, yes of course I can explain. generateStyles which takes three arguments: value, selected, and category. The purpose of this function is to generate a style object based on the input arguments. generateStyles function returns an object with three properties: backgroundColor, borderRadius, and border.
The function uses a switch statement to determine which category the style should be generated for. The three possible categories are "negative", "neutral", and "positive".
For each category, the function sets a backgroundColor and a borderColor based on the value argument 1 to 5, and word category.